ui/app/adapters/kv/metadata.js

```diff
@@ -3,39 +3,53 @@
  * SPDX-License-Identifier: MPL-2.0
  */
 
-import ApplicationAdapter from '../application';
 import { encodePath } from 'vault/utils/path-encoding-helpers';
-import { kvId } from 'vault/utils/kv-id';
+import ApplicationAdapter from '../application';
+import { kvMetadataPath } from 'vault/utils/kv-path';
 
 export default class KvMetadataAdapter extends ApplicationAdapter {
   namespace = 'v1';
 
-  _urlForMetadata(backend, path) {
-    return `${this.buildURL()}/${encodePath(backend)}/metadata/${encodePath(path)}`;
+  _url(fullPath) {
+    return `${this.buildURL()}/${fullPath}`;
   }
 
   createRecord(store, type, snapshot) {
     const { backend, path } = snapshot.record;
-    const url = this._urlForMetadata(backend, path);
-
-    return this.ajax(url, 'POST', { data: this.serialize(snapshot) }).then((resp) => {
-      resp.id = kvId(backend, path, 'metadata');
-      return resp;
+    const id = kvMetadataPath(backend, path);
+    const url = this._url(id);
+    const data = this.serialize(snapshot);
+    return this.ajax(url, 'POST', { data }).then(() => {
+      return {
+        id,
+        data,
+      };
     });
   }
 
-  queryRecord(store, type, query) {
-    const { path, backend } = query;
-    return this.ajax(this._urlForMetadata(backend, path), 'GET').then((resp) => {
-      resp.id = kvId(backend, path, 'metadata');
+  // TODO: replace this with raw request for metadata request?
```

Comment:
I ran into this question when looking at the results of the metadata list. For example, if you have two secrets `some-secret` and `some-secret/nested` and you query, the API returns `some-secret` and `some-secret/` separately. On the UI, clicking one of them links to the secret view and the other goes further in the tree. Seems like trying to use ember-data here would be shimming in something when really we only need the keys and whether it ends in `/`